File: Examples/StateDiagramDiff/Java/Before/Download.java
package net;

/** A download whose {@code state} advances through a pipeline: {@code run()} walks the happy
 *  path as a sequence of assignments (a transition chain), while {@code fail()} branches. */
public final class Download {
    private DownloadState state = DownloadState.IDLE;

    public void run() {
        state = DownloadState.REQUESTED;
        state = DownloadState.DOWNLOADING;
        state = DownloadState.FINISHED;
    }

    public void fail() {
        state = DownloadState.FAILED;
    }
}

File: Examples/StateDiagramDiff/Python/Before/download.py
from enum import Enum


class DownloadState(Enum):
    """The phases a download moves through."""

    IDLE = "idle"
    REQUESTED = "requested"
    DOWNLOADING = "downloading"
    VERIFYING = "verifying"
    FINISHED = "finished"
    FAILED = "failed"


class Download:
    """A download whose ``state`` advances through a pipeline: ``run`` walks the happy path as a
    sequence of assignments (a transition chain), while ``fail`` branches from the start."""

    state: DownloadState = DownloadState.IDLE

    def run(self) -> None:
        self.state = DownloadState.REQUESTED
        self.state = DownloadState.DOWNLOADING
        self.state = DownloadState.FINISHED

    def fail(self) -> None:
        self.state = DownloadState.FAILED
